src/ui/map.js

```diff
@@ -877,6 +879,25 @@ class Map extends Camera {
         return this;
     }
 
+    once(type: MapEvent, layerId: any, listener: any) {
+
+        if (listener === undefined) {
+            return super.once(type, layerId);
+        }
+
+        const delegatedListener = this._createDelegatedListener(type, layerId, listener);
+
+        this._delegatedListeners = this._delegatedListeners || {};
+        this._delegatedListeners[type] = this._delegatedListeners[type] || [];
+        this._delegatedListeners[type].push(delegatedListener);
```

Comment:
This function uses `Evented#once` to add listeners, which means that the event listener will be added to `this._oneTimeListeners`. Once the event is fired, it will be removed from that list. However, it will _not_ be removed from this._delegatedListeners`, which means that handlers are going to accumulate in this array causing memory growth.